challenge_two/src/agent.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  getJsonRpcUrl, ethers
+} from "forta-agent";
+
+
+import { SWAP_EVENT } from "./utils/constants";
+import { getPoolAddress } from "./utils/getPoolAddress";
+import { createFinding } from "./utils/createFinding";
+
+
+
+export const provideHandleTransaction =
+  (provider: ethers.providers.JsonRpcProvider): HandleTransaction =>
+  async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    // filter only for emitted swap events
+    const swapEvents = txEvent.filterLog(SWAP_EVENT);
+
+    // now the hard part is making sure they are uniswap swaps only
+
+    // How do I make sure the swap calls are to Uniswap and not other protocols with the same function signature?
+
+    for (const swap of swapEvents) {
+      try {
+        const computedAddress = await getPoolAddress(swap.address, provider);
+   
+        if (computedAddress.toLowerCase() != swap.address.toLocaleLowerCase()) {
+          return findings;
+        }
+        findings.push(createFinding(swap.address));
+        return findings;
```

Comment:
didn't try but this looks like the problem to your question. if you return in the middle of the loop, it wouldn't continue. you need to change this logic so that it can check every swapEvents